avocado/core/loader.py

```diff
@@ -195,6 +197,14 @@ def _str_loaders():
                 extra_params['loader_options'] = loaders[i][1]
             plugin = self.registered_plugins[supported_loaders.index(name)]
             self._initialized_plugins.append(plugin(args, extra_params))
+        # Plugins are initialized, let's update mappings
+        self._label_mapping = {test.MissingTest: "MISSING"}
+        for plugin in self._initialized_plugins:
+            self._label_mapping.update(plugin.get_type_label_mapping())
+        self._decorator_mapping = {test.MissingTest:
+                              output.TERM_SUPPORT.fail_header_str}
+        for plugin in self._initialized_plugins:
+            self._decorator_mapping.update(plugin.get_decorator_mapping())
```

Comment:
You moved the code from `get_type_label_mapping()`/`get_decorator_mapping()` to `load_plugins()`. That's the part I find unnecessary/bad-fit. Creating new private functions is not my intention.